stubs/python-dateutil/dateutil/rrule.pyi

```diff
@@ -25,21 +56,21 @@ SA: weekday
 SU: weekday
 
 class rrulebase:
-    def __init__(self, cache: bool = False) -> None: ...
+    def __init__(self, cache: bool | None = False) -> None: ...
     def __iter__(self) -> Iterator[datetime.datetime]: ...
     def __getitem__(self, item): ...
-    def __contains__(self, item): ...
-    def count(self): ...
+    def __contains__(self, item) -> bool: ...
+    def count(self) -> int | None: ...
     def before(self, dt, inc: bool = False): ...
     def after(self, dt, inc: bool = False): ...
-    def xafter(self, dt, count=None, inc: bool = False): ...
-    def between(self, after, before, inc: bool = False, count: int = 1): ...
+    def xafter(self, dt, count=None, inc: bool = False) -> Generator[Incomplete]: ...
+    def between(self, after, before, inc: bool = False, count: int = 1) -> list[Incomplete]: ...
 
 class rrule(rrulebase):
     def __init__(
         self,
-        freq,
-        dtstart: datetime.date | None = None,
+        freq: Literal[0, 1, 2, 3, 4, 5, 6],
+        dtstart: datetime.datetime | None = None,
```

Comment:
The old annotation was correct. `rrule` converts a `date` to a `datetime`:

https://github.com/dateutil/dateutil/blob/35ed87a02ad777428a9da84ead4d8425b533d4a9/src/dateutil/rrule.py#L441

I'd also be fine with using `datetime.datetime | datetime.date`. Although technically redundant, it's a bit clearer. (Also deriving `datetime` from `date` was a mistake, in my opinion.)

See also the prefect primer hits.